templates/repo/diff/compare.tmpl

```diff
@@ -184,23 +184,18 @@
 		{{end}}
 	{{else if and .PageIsComparePull (gt .CommitCount 0)}}
 		{{if .HasPullRequest}}
-			<div class="ui segment grid title">
-				<div class="twelve wide column issue-title">
-					{{ctx.Locale.Tr "repo.pulls.has_pull_request" (print $.RepoLink "/pulls/" .PullRequest.Issue.Index) $.RepoRelPath .PullRequest.Index}}
-					<h1>
-						<span id="issue-title">{{RenderIssueTitle $.Context .PullRequest.Issue.Title ($.Repository.ComposeMetas ctx)}}</span>
+			<div class="existing-issue-block ui segment tw-flex tw-items-center tw-space-between tw-p-3 tw-gap-3">
+				<div class="tw-flex tw-flex-1 tw-gap-3 tw-items-center gt-ellipsis">
+					{{template "shared/issuestatelabel" (dict "." . "Issue" .PullRequest.Issue)}}
```

Comment:
Please pass the dict keys/values explicitly for what the sub-template really needs.

![image](https://github.com/go-gitea/gitea/assets/2114189/2147e586-1a9e-4fa2-bf63-cdfb615ded70)

Reply:
This is obsolete, it seems to work with two times `{{template "shared/issueicon" .}}` currently.

Reply:
I suppose. BTW is there a difference between 

```
{{template "shared/issueicon" dict "Issue" .}}
```

and 

```
{{template "shared/issueicon" (dict "Issue" .)}}
```

I see both forms in use.

Reply:
IIRC no much difference actually. Just like `1+2` and `(1+2)` in other languages.

Reply:
The problem is the subtemplate expects `.PullRequest` at root, so I would need to refactor all call sites...

Reply:
I guess I could split out all the accessed properties...

Reply:
Imho this whole `.Issue.PullRequest` and `.PullRequest.Issue` business is ripe for a major refactor in the backend where it just merges both into the same struct and differentiates via `.isPull`.

Reply:
If `shared/issueicon` is from legacy code, then it's fine to leave it as-is. It is out of this PR's scope.

Reply:
Yeah, the untangling of issue/pr is something for later. A fix here would be 

```
{{template "shared/issueicon" dict "Issue" .PullRequest.Issue "PullRequest" .PullRequest}}
```

At all call sites, but I don't think this is worth currently without also untangling this mess in the backend.